cis_proc.py

```diff
@@ -259,60 +256,84 @@ def main(argv=None):
+                    run_mriqc = True
                 else:
-                    run_mriqc=False
+                    run_mriqc = False
             else:
-                if op.isfile(op.join(scratch_bids_dir, 'sub-{sub}'.format(sub=args.sub), 'func/sub-{sub}_task-{task}_run-01_bold.json'.format(sub=args.sub, task=tmp_task))):
-                    run_mriqc=True
+                task_json_file = op.join(
+                    scratch_bids_dir,
+                    'sub-{sub}/func/sub-{sub}_task-{task}_run-01_bold.'
+                    'json'.format(sub=sub, task=tmp_task))
+                if op.isfile(task_json_file):
+                    run_mriqc = True
                 else:
-                    run_mriqc=False
-            
-            if run_mriqc:   
-                kwargs = ''
+                    run_mriqc = False
+
+            if run_mriqc:
+                kwarg_str = ''
                 for field in config_options['mriqc_options']['func']['task'][tmp_task]['mriqc_settings'].keys():
                     if isinstance(config_options['mriqc_options']['func']['task'][tmp_task]['mriqc_settings'][field], list):
                         val = ' '.join(config_options['mriqc_options']['func']['task'][tmp_task]['mriqc_settings'][field])
                     else:
                         val = config_options['mriqc_options']['func']['task'][tmp_task]['mriqc_settings'][field]
-                    kwargs += '--{0} {1} '.format(field, val)
-                kwargs = kwargs.rstrip()
-                cmd = ('singularity run --cleanenv -B {templateflowdir}:$HOME/.cache/templateflow {sing} {bids} {out} participant --no-sub --verbose-reports '
+                    kwarg_str += '--{0} {1} '.format(field, val)
+                kwarg_str = kwarg_str.rstrip()
+                cmd = ('singularity run --cleanenv '
+                       '-B {templateflowdir}:$HOME/.cache/templateflow '
+                       '{sing} {bids} {out} participant '
+                       '--no-sub --verbose-reports '
                        '--task-id {task} -m bold '
                        '-w {work} --n_procs {n_procs} --correct-slice-timing '
-                       '{kwargs} '.format(templateflowdir = op.join(args.work_dir, 'templateflow'), sing=scratch_mriqc, bids=scratch_bids_dir,
-                                          out=scratch_deriv_dir, task=tmp_task,
-                                          work=mriqc_work_dir, n_procs=n_procs, kwargs=kwargs))
+                       '{kwarg_str}'.format(
+                            templateflowdir=op.join(work_dir, 'templateflow'),
+                            sing=scratch_mriqc,
+                            bids=scratch_bids_dir,
+                            out=scratch_deriv_dir,
+                            task=tmp_task,
+                            work=mriqc_work_dir,
+                            n_procs=n_procs,
+                            kwarg_str=kwarg_str))
                 run(cmd)
```

Comment:
I think this entire MRIQC section needs a lot of love. I think we should create some kind of parser for `mriqc_config` to offload some of this work. It's very hard to read and understand everything that's happening at present.

Reply:
I'm planning to split all of the different steps into separate functions and reorganize the workflow into a simple package. This can happen at that point.